### DISCUSSION TITLE
Deadlock on simultaneous transactions

### DISCUSSION THREAD

**Discussion by ggwpez** [hyperledger-labs/go-perun#32] [Q&A]
Since we require a fully signed state for State Apps, they can deadlock when both parties propose a new state to each other at the same time.  
This can also happen when both parties close at the same time, we we already observed in the mobile app.  

Lets discuss on how to circumvent this; Some kind of priority would be possible. A client would always gives the lower `Actor Index` precedence.  
Other ideas?  
@matthiasgeihs

**Comment by matthiasgeihs** [comment]
Can you give more detail on where the deadlock happens?

I think the natural way of handling this would be via a timeout.

**Reply by manoranjith** [reply]
In `perun-node` as well, we have use a `reponseTimeout` for off-chain messages in order to prevent this kind of dead lock.

**Reply by ggwpez** [reply]
The experience for the user is not really different with a timeout, since the user then has to wait for the timeout instead of the deadlock.  
I would favor a protocol solution which actually solves the problem for any Channel regardless of timing.
